### DISCUSSION TITLE
Will Grouping and other numeric functions be implemented?

### DISCUSSION THREAD

**Discussion by ggarnhart** [supabase/supabase#1560] [Questions] [answered]
Hi friends!

I've loved using Supabase -- the API is friendly and the documentation is great. Is there a way to run an `AVG()` function on a numeric column? If so, I'd love to learn! If not, I can always implement that on my end, but figured I'd ask to see if that's something that's happened yet.

I'd also love to learn about Group By functions and whether or not that is expected in the future.

**Comment by soedirgo** [comment, marked as answer]
Hey @ggarnhart, this is currently not supported (the tracking issue on PostgREST side is here https://github.com/PostgREST/postgrest/issues/915), so for now you'll have to do this through stored procedures/views.

**Reply by awalias** [reply]
just to elaborate on the current recommended approach:

- write a function and call it from the API: https://supabase.io/docs/reference/javascript/rpc
- write a view and query it as normal: https://www.tutorialspoint.com/postgresql/postgresql_views.htm

**Reply by ggarnhart** [reply]
Great! Thanks team :)
